Mathlib/Analysis/Normed/Order/UpperLower.lean

```diff
@@ -154,3 +198,63 @@ theorem IsLowerSet.exists_subset_ball (hs : IsLowerSet s) (hx : x ∈ closure s)
 #align is_lower_set.exists_subset_ball IsLowerSet.exists_subset_ball
 
 end Fintype
+
+section Finite
+variable [Finite ι] {s t : Set (ι → ℝ)} {a₁ a₂ b₁ b₂ x y : ι → ℝ} {δ : ℝ}
+
+/-!
+#### Note
+
+The closure and frontier of an antichain might not be antichains. Take for example the union
+of the open segments from `(0, 2)` to `(1, 1)` and from `(2, 1)` to `(3, 0)`. `(1, 1)` and `(2, 1)`
+are comparable and both in the closure/frontier.
+-/
+
+protected lemma IsClosed.upperClosure (hs : IsClosed s) (hs' : BddBelow s) :
+    IsClosed (upperClosure s : Set (ι → ℝ)) := by
+  cases nonempty_fintype ι
+  refine' IsSeqClosed.isClosed fun f x hf hx ↦ _
+  choose g hg hgf using hf
+  obtain ⟨a, ha⟩ := hx.bddAbove_range
+  obtain ⟨b, hb, φ, hφ, hbf⟩ := tendsto_subseq_of_bounded (hs'.isBounded_inter bddAbove_Iic) fun n ↦
+    ⟨hg n, (hgf _).trans $ ha $ mem_range_self _⟩
```

Comment:
https://leanprover-community.github.io/contribute/style.html#whitespace-and-delimiters says to use `<|`, not `$`.